VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -187,7 +186,17 @@ public static CharSequence trim(CharSequence s) {
     }
 
     public static CharSequence removeDuplicateWhitespace(CharSequence text) {
-        return PATTERN_DUPLICATE_WHITESPACE.matcher(text).replaceAll(" ");
+        SpannableStringBuilder builder = new SpannableStringBuilder(text);
+        for (int size = text.length(), i = 0; i < size; i++) {
+            if (Character.isWhitespace(builder.charAt(i))) {
+                int nextIndex = i + 1;
+                if (nextIndex < size && Character.isWhitespace(builder.charAt(nextIndex))) {
+                    builder.replace(i, nextIndex, " ");
+                    break;
+                }
+            }
```

Comment:
Here is my alternate code for the contents of this loop, which replaces any number of duplicates in one go, and resolved the other issues I commented on as well:

```` java
int whitespaceStartIndex = i;
for (; builder.charAt(i) == ' '; i++);
if (i > whitespaceStartIndex + 1) {
    builder.replace(whitespaceStartIndex, i + 1, " ");
}
````